openquake/hazardlib/source/simple_fault.py

```diff
@@ -160,7 +181,8 @@ def iter_ruptures(self):
                     mesh = whole_fault_mesh[first_row: first_row + rup_rows,
                                             first_col: first_col + rup_cols]
 
-                    if self.hypo_list.size == 1:
+                    if ((self.hypo_list.size == 1) and
```

Comment:
hypo_list.size =  1 will exist only when hypo_list=numpy.array(None), which is the default meaning no hypocentre location is given by user. 

Once we give the hypo_list, for example, hypo_list=numpy.array([0.5, 0.5, 1]), the hypo_list.size will be 3. By the size of the array, it allows us to distinguish if hypo_list is given or not.

If we use len(hypo_list), it will give error when hypo_list = numpy.array(None) as "TypeError: len() of unsized object"